test/show.jl

```diff
@@ -1263,7 +1263,7 @@ let repr = sprint(dump, :(x = 1))
     @test repr == "Expr\n  head: Symbol =\n  args: Array{Any}((2,))\n    1: Symbol x\n    2: $Int 1\n"
 end
 let repr = sprint(dump, Pair{String,Int64})
-    @test repr == "Pair{String, Int64} <: Any\n  first::String\n  second::Int64\n"
```

Comment:
So all non mutable structs will print `const` for all their fields? Isn't it better to leave the non-mutable structs as is and only print out `const` for mutable structs with const fields (together with adding `mutable` in the dump for those types).

Reply:
You're right, that might be strange. I've now made the following changes:
* Print the type of type (`struct`, `primitive type` etc.) at the beginning before the name
* Don't print `const` unless in a mutable struct.

Here are some examples:
```
julia> dump(Float32)
primitive type Float32 <: AbstractFloat

julia> dump(Signed)
abstract type Signed <: Integer

julia> dump(Pair{Int, String})
struct Pair{Int64, String} <: Any
  first::Int64
  second::String

julia> dump(BitSet)
mutable struct BitSet <: AbstractSet{Int64}
  const bits::Vector{UInt64}
  offset::Int64

julia> dump(Set)
UnionAll
  var: TypeVar
    name: Symbol T
    lb: Union{}
    ub: abstract type Any
  body: struct Set{T} <: AbstractSet{T}
    dict::Dict{T, Nothing}
```

Reply:
👌